projects/rpx-xui-translation/src/lib/rpx-translation.service.ts

```diff
@@ -153,18 +117,26 @@ export class RpxTranslationService {
 
     this.requestTimerSubscription = timer(this.config.debounceTimeMs).subscribe(() => {
       const url = this.config.baseUrl + `/${lang}`;
-      const s = this.http.post<TranslationsDTO>(url, { phrases: this.requesting[lang] }).pipe(
-        map(t => t.translations),
-        catchError(() => {
-          const translations: { [from: string]: string } = {};
-          this.requesting[lang].forEach(p => translations[p] = this.config.testMode ? `[Test translation for ${p}]` : p);
-          return of(translations);
-        })
-      ).subscribe((translations: { [from: string]: TranslationModel }) => {
+      const s = this.http.post<TranslationsDTO>(url,
+        {
+          phrases: this.requesting[lang],
+        }
+      )
+        .pipe(
+          map(t => t.translations),
+          catchError(() => {
+            const translations: { [from: string]: string } = {};
+            this.requesting[lang].forEach(p => translations[p] = this.config.testMode ? `[Test translation for ${p}]` : p);
+            return of(translations);
+          })
+      ).subscribe((translations: { [from: string]: string }) => {
         const toAdd: Translation[] = [];
         Object.keys(translations).forEach(p => {
-          toAdd.push(Translation.create(p, lang, translations[p], DateTime.now().plus(this.config.validity).toISO()));
-          this.phrases[p].next(this.getPhrase(translations[p], yesOrNo));
+          const translation: TranslatedData = {
```

Comment:
you already have the TranslatedData models here to store (translations is an array of phrase -> TranslatedData), so no need to wrap it inside another object.

Reply:
API doesn't seem to return this. It currently returns { [from: string]: string }